### DISCUSSION TITLE
How to factor fractions?

### DISCUSSION THREAD

**Discussion by n-gram-hub** [sympy/sympy#25878] [Q&A] [answered]
I'm trying to factor polynomials, but I couldn't get the right result with fractions.

Example:

9/4y^2 - 49/81z^2

The expected result would be:

(3/2y - 7/9z)*(3/2y + 7/9z)

but with this code

`print(factor(Fraction(9/4).limit_denominator()*y**2 - Fraction(49/81).limit_denominator()*z**2))`

and this one too:

`print(factor(y**2*Rational(9,4) - z**2*Rational(49, 81)))`

I get:

(27*y - 14*z)*(27*y + 14*z)/324

Is there any way to get (3/2y - 7/9z)*(3/2y + 7/9z)?

Thanks, any help is appreciated

**Comment by oscarbenjamin** [comment, marked as answer]
I don't see how to generalise from your example. There are infinitely many possible ways to multiply one factor and divide another. How would the result be uniquely defined?

**Reply by n-gram-hub** [reply]
@oscarbenjamin Well, that's ok, but this is a quite typical scenario where you have a sum and difference product. Maybe I thought Sympy could automatically do that because it works perfectly when you haven't any fractions (eg. `print(factor('x**2-16'))`). Thanks for your kind reply

**Reply by oscarbenjamin** [reply]
The factor function outputs results in a canonical form by factoring out the denominator and then the gcd of the coefficients to get a primitive polynomial with integer coefficients. It then factorises that into primitive factors. Afterwards you can recombine the numbers factored out in different ways.

The algorithms involved here are significantly more complicated than recognising a difference of squares. It might be useful to have a different function for that though.

**Reply by n-gram-hub** [reply]
@oscarbenjamin Please, add a dedicated function   :-)
